Feature/jquery less

guzmandrade-dev opened the pull request:
This PR refactor jQuery code with vanilla JavaScript, trying to preserve as much as possible the original functionality.

The most important change is:

https://github.com/WebDevStudios/Hash-Link-Scroll-Offset/pull/20/files#diff-dac915876df5b3fd289787b814b8677224bdf09c3feceffd4a757ec531b8d213R125-R136

where the scroll animation now uses `window.scrollTo` instead of jQuery `animate()`.

Also, this PR replaces `Grunt` with `@wordpress/scripts` to minify the JavaScript code, where we are keeping the same enqueue URL as `/assets/js/hash-link-scroll-offset.js`, but the source file changed to `/src/hash-link-scroll-offset.js` and we are no longer generating a `.min.js` file.

Commit:
de-jQuering code and replacing functionality with vanilla JavaScript

Commit:
grunt files

Commit:
Merge branch 'master' into feature/jquery-less

Commit:
de-grunt plugin and replace minification with `@wordpress/scripts`

Commit:
code formatting

tw2113 commented:
Some initial thoughts/notes:

1. in the build directory, we should still have a non-minified copy and a minified copy of the js. So the SCRIPT_DEBUG parts would remain and whatnot.
2. Did you intentionally change the shorthand arrays to longerhand?
3. Let's get the `$html_and_body` renamed without the $ as that's typically meant to indicate the variable is a jQuery object, which in this case it wouldn't be any more.

Commit:
custom webpack configuration to copy source file

Commit:
short array syntax

Commit:
adding SCRIPT_DEBUG back to conditionally enqueue minified script

Commit:
rename $html_and_body to camelCase

guzmandrade-dev commented:
> Some initial thoughts/notes:
> 
> 1. in the build directory, we should still have a non-minified copy and a minified copy of the js. So the SCRIPT_DEBUG parts would remain and whatnot.
> 2. Did you intentionally change the shorthand arrays to longerhand?
> 3. Let's get the `$html_and_body` renamed without the $ as that's typically meant to indicate the variable is a jQuery object, which in this case it wouldn't be any more.

Thank you.

1. Since the code can create a `.map` file when running `npm start` I omitted that `SCRIPT_DEBUG` condition. It is back now. 
2. Yes, I intentionally change the shorthand array syntax when formatting the code with PHPCS. I reverted that change.
3. I renamed `$html_and_body` to be camelCase instead.
